src/hooks/use-arrow-key-navigation.ts

```diff
@@ -82,9 +87,17 @@ export function useArrowKeyNavigation(
       const elements: HTMLElement[] = Array.from(
         container.querySelectorAll(selector)
       );
-      return elements.filter(
+      const filtered = elements.filter(
         el => isElementVisible(el) && !isElementDisabled(el)
       );
+      // Include the container itself in the set of navigable elements if it
+      // is currently focused. It will not be part of the tab sequence once it
+      // loses focus. This allows, e.g., a modal container to be focused when
+      // opened but not be part of the subsequent trapped tab sequence.
```

Comment:
This comment was directly pasted from `useTabKeyNavigation` and isn't quite applicable. I'll adjust after review.